ERCS/erc-7589.md

```diff
@@ -0,0 +1,480 @@
+* The `revokeRole` function MAY be implemented as public or external and MUST revert if:
+  * The role assignment is not found (no role was granted).
+  *  The `msg.sender` was not approved by the grantor or the grantee.
+  *  The `msg.sender` is the grantor or was approved by the grantor, but the role is not revocable or expired.
+* The `releaseTokens` function MAY be implemented as public or external and MUST revert if:
+  * The commitment is not found (no tokens were committed).
+  * The `msg.sender` is not and was not approved by the grantor.
+  * The commitment has at least one non-revocable role that didn't expire.
+* The `setRoleApprovalForAll` function MAY be implemented as public or external.
+* The `grantorOf` function MAY be implemented as pure or view and MUST return the owner of the committed tokens.
+* The `tokenAddressOf` function MAY be implemented as pure or view and MUST return the address of the committed tokens.
+* The `tokenIdOf` function MAY be implemented as pure or view and MUST return the identifier of the committed tokens.
+* The `tokenAmountOf` function MAY be implemented as pure or view and MUST return the token amount committed.
+* The `roleData` function MAY be implemented as pure or view and MUST return the custom data of the role assignment.
+* The `roleExpirationDate` function MAY be implemented as pure or view and MUST return the expiration date of the role
+  assignment.
+* The `isRoleRevocable` function MAY be implemented as pure or view and MUST return whether the grantor can end the role
+  assignment before its expiration date.
+* The `isRoleApprovedForAll` function MAY be implemented as pure or view and MUST return whether the `_operator` is
+  allowed to grant and revoke roles on behalf of the `_grantor`.
+
+> Please note that "approval" refers to allowing users to commit tokens and grant/revoke roles on one's behalf. An
+  approved user either received the role approval or is the target user. Role approvals are not to be confused with
+  [ERC-1155](./eip-1155.md) approvals. More information can be found in the [Role Approvals](#role-approvals) section.
+
+## Rationale
+
+The conventional methods for utilizing SFTs often involve transferring tokens to another address or freezing balances to
+restrict transfers. However, the question arises: How can SFT owners securely delegate the utility of their assets
+without compromising custody?
+
+[ERC-7589](./eip-7589.md) steps in as a solution by introducing the concept of "token commitments". This abstraction
+serves as a powerful tool for users looking to delegate the control of their SFTs. A token commitment can represent
+either a frozen balance or tokens deposited into a contract, offering a standardized and secure way for SFT owners to
+delegate the use of their assets. Through ERC-7589, users gain a versatile mechanism to abstract the complexities of
+secure delegation, enhancing the utility and interoperability of semi-fungible tokens.
```

Comment:
These two paragraphs look like they belong in Motivation, since they're justifying the proposal as a whole. Your Rationale section should explain technical choices made _within_ the proposal.